android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -1033,6 +1035,10 @@ class GeckoEngineSession(
             notifyObservers { onCookieBannerChange(CookieBannerHandlingStatus.HANDLED) }
         }
 
+        override fun onProductUrl(session: GeckoSession) {
```

Comment:
For my own understanding, can this function actually get the boolean `isProductUrl` and that's passed on to `onProductUrlChange`. As we are not doing anything with the session, we might not need to pass that and instead just pass the boolean. As a result we could also remove the reseting of the status on lines 710-711, wdyt?

Reply:
This is the GV API. The way it works, this is part of a `ContentDelegate` interface. GV calls this function once a product URL has been detected. The way to use the API is you override the interface method (as I am doing here) and you write your code, whatever you want to happen when a product URL is encountered.

We are passing in the GeckoSession to make it available to people in case they need it (we have multiple APIs like this, see `onCookieBannerHandled` API right above). If you don't need it - it's fine, you don't have to use it. ("You" as in "the consumer of GV API", so in this case the AC code)

Reply:
This is the GV API. The way it works, this is part of a `ContentDelegate` interface. GV calls this function once a product URL has been detected. The way to use the API is you override the interface method (as I am doing here) and you write your code, whatever you want to happen when a product URL is encountered.

We are passing in the GeckoSession to make it available to people in case they need it (we have multiple APIs like this, see `onCookieBannerHandled` API right above). If you don't need it - it's fine, you don't have to use it. ("You" as in "the consumer of GV API", so in this case the AC code)

Reply:
(to add to the explanation, GV is a general Android library, it's not Fenix-specific or AC-specific)

Reply:
I understand that.

What I'm saying is if the GV API was something like this:

`fun onProductUrl(isProductUrl: Boolean)`

So we won't need to reset as we are doing above and the api returns what we needs, not `GeckoSession`.

This is an idea and with/without it doesn't impact the consumer of AC, happy to discuss more on slack.

Reply:
Sure, we can always discuss on Slack! I am resolving this for now.